com.avaloq.tools.ddk.checkcfg.core/src/com/avaloq/tools/ddk/checkcfg/validation/CheckCfgJavaValidator.java

```diff
@@ -306,10 +307,25 @@ public String apply(final ConfiguredCheck from) {
   @Check
   public void checkConfiguredLanguageExists(final ConfiguredLanguageValidator validator) {
     if (!checkCfgUtil.getAllLanguages().contains(validator.getLanguage())) {
-      error("Unknown language", validator, CheckcfgPackage.Literals.CONFIGURED_LANGUAGE_VALIDATOR__LANGUAGE, ValidationMessageAcceptor.INSIGNIFICANT_INDEX, IssueCodes.UNKNOWN_LANGUAGE);
+      error("Unknown language", validator, CheckcfgPackage.Literals.CONFIGURED_LANGUAGE_VALIDATOR__LANGUAGE, ValidationMessageAcceptor.INSIGNIFICANT_INDEX, IssueCodes.UNKNOWN_LANGUAGE); //$NON-NLS-1$
     }
   }
 
+  /**
+   * Checks that language referenced in validator configuration is unique.
+   *
+   * @param checkCfg
+   *          the check configuration containing {@link ConfiguredLanguageValidator}'s
+   */
+  @Check
+  public void checkConfiguredLanguageUnique(final CheckConfiguration checkCfg) {
+    Set<String> existingLanguages = Sets.newHashSet();
+    checkCfg.getLanguageValidatorConfigurations() //
+        .stream() //
+        .filter(lang -> !existingLanguages.add(lang.getLanguage())) // Don't do the mapping first - it's easier having the EObject

```

Comment:
Not quite so simple -- you need to check that either the parameters or the configured catalogs match before you can say they're not unique.